content/blog/2025/07/2025-07-16-security-predisclosure.md

```diff
@@ -0,0 +1,33 @@
++++
+date = "2025-07-16"
+title = "Pre-disclosure: Upcoming coordinated security fix for all Matrix server implementations"
+
+[taxonomies]
+author = ["Matthew Hodgson"]
+category = ["General", "Security"]
++++
+
+Hi all,
+
+Over the last 6 months a major project has been underway by the Element server team and the Matrix.org Foundation security team to investigate “state resets”: scenarios where Matrix’s state resolution algorithm can give unexpected results.  As part of this work we’ve identified two high severity protocol vulnerabilities (CVE-2025-49090; the other not yet allocated a CVE).
+
+Given the security implications of a federation protocol vulnerability, we’ve shared details under embargo over the last 4 weeks with all known active server implementations, and are now aiming for a coordinated security release across all Matrix server implementations on **Tuesday Jul 22nd 2025 at 17:00 UTC**.  If you run a Matrix server in an untrusted federation (e.g. the public federation), please be prepared to upgrade as soon as the patched versions are available.
+
```

Comment:
Would you be opposed to adding the teaser marker? This L15 feels like it could be the appropriate place for it to me.

You can find the docs on how to do so in the standard places of the repo as well as linked (in the comment form only visible to the PR creator) in the PR template.

Reply:
no, i don't want a teaser marker, the whole post is critical.